Header header
bool positionReached #TODO: NYI
